### DISCUSSION TITLE
I need help with Arduino Mega OTA

### DISCUSSION THREAD

**Discussion by heinsschonhals** [JAndrassy/ArduinoOTA#255] [General]
Hello, I am getting the following feedback when I send a sketch via the network port:

"Connecting to board... done
Uploading sketch...done
Flashing sketch... Error flashing the sketch"

I have already changed the timeout but it may not be correct, as this error message comes in approximately 10 seconds, but the following command is given before this message appears:

"C:\Users\heins\AppData\Local\Arduino15\packages\arduino\tools\arduinoOTA\1.3.0/bin/arduinoOTA -address 192.168.3.6 -port 65280 -username arduino -password heins -sketch C:\Users\ heins\AppData\Local\Temp\arduino_build_653848/testeOTA.ino.bin -upload /sketch -b -t 60"

I installed the Megacore bootloader "stk500boot_v2_atmega2560_115200_16000000L.hex", if it's not the correct one, if you could send me the correct bootloader, I've also replaced the arduinoOTA.exe file with version 1.4.1, I think it was in 1.3.0

**Comment by JAndrassy** [comment]
you need optiboot botloader. not stl500v2

**Comment by heinsschonhals** [comment]
> você precisa do botloader optiboot. não stl500v2

ok, I managed to make sending the sketch to the network port work, but I need to restart the Arduino manually for the new sketch to work, any solution?

**Comment by heinsschonhals** [comment]
> > em qual arquivo e qual função chama o watchdog, estou procurando isso agora
> 
> copy_flash_pages no bootloader

I'm using a set of optiboot 8 cards, maybe that's it, but it shouldn't work when I restart manually, right?

**Comment by heinsschonhals** [comment]
I use the following configuration
![optiboot](https://github.com/JAndrassy/ArduinoOTA/assets/120427725/3f28cbb9-e9d8-47f8-a3bb-7aa6ec8d6ace)

**Comment by JAndrassy** [comment]
that doesn't have copy_flash_pages. use MegaCore version or my_boards

btw: the "and after manual reset the new sketch runs?" question was for mcerqueira1509

**Comment by heinsschonhals** [comment]
> que não tem copy_flash_pages. use a versão MegaCore ou my_boards
> 
> aliás: o "e após a redefinição manual o novo esboço é executado?" a pergunta era para mcerqueira1509

I found it in your repository, later I will test it and give feedback here if it works to help more people

![image](https://github.com/JAndrassy/ArduinoOTA/assets/120427725/def5645e-8390-4435-b56a-6ff9d49db900)

**Comment by heinsschonhals** [comment]
Another question, if you have already tested it, sending the sketch outside the local network, through a public IP, would there be a manual way to add the network port to the IDE?

**Comment by JAndrassy** [comment]
yes. read the README

**Comment by heinsschonhals** [comment]
> sim. leia o README

bingo worked, thank you very much for your help, I will study public IP, this readme from the library folder?

**Comment by JAndrassy** [comment]
here https://github.com/JAndrassy/ArduinoOTA

**Comment by heinsschonhals** [comment]
> aqui https://github.com/JAndrassy/ArduinoOTA

ok in theory this worked, I need to validate it while still outside the same network, I did it with my public IP and it worked, not the first time because instead of pulling the password I have in the sketch, it pulled the default password "password" and then it gave as access denied, even though the dialog box to enter the password is not opened

**Comment by peno64** [comment]
I must be doing something wrong...

I have an arduino mega 2560 R3

I can upload the blink sketch to it and it works with the default bootloader.

I burned the new bootloader from https://github.com/JAndrassy/my_boards/blob/master/avr/bootloaders/optiboot8/optiboot_atmega2560.hex
To be able to do this, I copied this file to C:\Users\peno64\AppData\Local\Arduino15\packages\Optiboot\hardware\avr\0.8.0\bootloaders\optiboot
It says success when I did this.
I appended the content of https://github.com/JAndrassy/my_boards/blob/master/avr/boards.txt to my boards.txt
In the IDE I choose board Optiboot 8.0 - Arduino Mega 2560 (Optiboot)
![image](https://github.com/user-attachments/assets/0eeed320-cd68-444e-9e70-9989c80dc756)
Now I upload, still via usb, the blink sketch. Upload hangs some time and then gives an error.

I can burn the default Arduino Mega bootloader again after this and everything is ok again.

What am I doing wrong?

Thank you

**Reply by JAndrassy** [reply]
maybe use my_boards this way
https://github.com/JAndrassy/my_boards/blob/master/my_boards.png

**Reply by peno64** [reply]
Thank you very much for your quick reply. That did it. Also note that I use Arduino IDE 2.3.2 but all works.

As a future reference do the following:

- go to https://github.com/JAndrassy/my_boards/tree/master and choose download zip
- extract the zip to C:\Users\<user>\AppData\Local\Arduino15\packages
- restart the IDE

This will add a bunch of my_boards-* items:

![image](https://github.com/user-attachments/assets/83161256-276a-4925-8c34-b0d385f91d6b)

- Choose from my_boards-master-avr (in Sketchbook), Arduino Mega 2560 (Optiboot)
- Choose the correct programmer
- Choose Burn Bootloader

**Reply by JAndrassy** [reply]
on the picture I sent you, my_boards is in sketchbook's hardware folder and your screenshot indicates that you put it there but your description has Arduino15/packages
